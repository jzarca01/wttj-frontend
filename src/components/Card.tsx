import { useSortable } from "@dnd-kit/sortable";

import { Task } from "../types";

type Props = {
  task: Task;
};

export const Card = ({ task }: Props) => {
  const { attributes, listeners, setNodeRef, transform, transition } =
    useSortable({
      id: task.id,
    });

  const style = transform
    ? {
        transform: `translate3d(${transform.x}px, ${transform.y}px, 0)`,
        transition,
      }
    : undefined;

  return (
    <div
      ref={setNodeRef}
      className="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
      style={style}
      {...listeners}
      {...attributes}
    >
      <h3 className="font-medium text-neutral-100">{task.title}</h3>
      <p className="mt-2 text-sm text-neutral-400">{task.description}</p>
    </div>
  );
};
