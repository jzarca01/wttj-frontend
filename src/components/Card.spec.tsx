import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Card } from "./Card";

import type { Task } from "../types";

describe("Card", () => {
  const mockTask: Task = {
    id: 1,
    title: "Test Task",
    description: "Test Description",
    status: "TODO",
  };

  it("renders the card with task title and description", () => {
    const result = render(<Card task={mockTask} />);
    expect(screen.getByText("Test Task")).toBeInTheDocument();
    expect(screen.getByText("Test Description")).toBeInTheDocument();
    expect(result).toMatchSnapshot();
  });
});
