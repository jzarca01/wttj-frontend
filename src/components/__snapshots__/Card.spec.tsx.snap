// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders the card with task title and description 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-describedby=""
        aria-disabled="false"
        aria-roledescription="sortable"
        class="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
        role="button"
        tabindex="0"
      >
        <h3
          class="font-medium text-neutral-100"
        >
          Test Task
        </h3>
        <p
          class="mt-2 text-sm text-neutral-400"
        >
          Test Description
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-describedby=""
      aria-disabled="false"
      aria-roledescription="sortable"
      class="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
      role="button"
      tabindex="0"
    >
      <h3
        class="font-medium text-neutral-100"
      >
        Test Task
      </h3>
      <p
        class="mt-2 text-sm text-neutral-400"
      >
        Test Description
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
